includes/internal/emails/class-email-repository.php

```diff
@@ -0,0 +1,193 @@
+	 * @param string $subject     Email subject.
+	 * @param string $description Email description.
+	 * @param string $content     Email content.
+	 *
+	 * @return bool
+	 */
+	public function create( string $identifier, array $types, string $subject, string $description, string $content ): bool {
+		if ( $this->has( $identifier ) ) {
+			return false;
+		}
+
+		$email_data = [
+			'post_status'  => 'publish',
+			'post_type'    => Email_Post_Type::POST_TYPE,
+			'post_author'  => 1,
+			'post_title'   => $subject,
+			'post_content' => $content,
+			'meta_input'   => [
+				self::META_IDENTIFIER  => $identifier,
+				self::META_DESCRIPTION => $description,
+			],
+		];
+
+		$email_id = wp_insert_post( $email_data );
+
+		foreach ( $types as $type ) {
+			add_post_meta( $email_id, self::META_TYPE, $type );
+		}
+
+		return $email_id > 0;
+	}
+
+	/**
+	 * Delete email for identifier.
+	 *
+	 * @internal
+	 *
+	 * @param string $identifier Email identifier.
+	 * @return bool
+	 */
+	public function delete( string $identifier ): bool {
+		$query = new WP_Query(
+			[
+				'select'     => 'ID',
+				'post_type'  => Email_Post_Type::POST_TYPE,
+				'meta_key'   => self::META_IDENTIFIER, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+				'meta_value' => $identifier, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+			]
+		);
+
+		if ( 0 === $query->post_count ) {
+			return false;
+		}
+
+		$email_id = $query->posts[0]->ID;
+
+		return (bool) wp_delete_post( $email_id, true );
+	}
+
+	/**
+	 * Get email for identifier.
+	 *
+	 * @internal
+	 *
+	 * @param string $identifier Email identifier.
+	 * @return \WP_Post|null
+	 */
+	public function get( string $identifier ) {
+		$query = new WP_Query(
+			[
+				'post_type'  => Email_Post_Type::POST_TYPE,
+				'meta_key'   => self::META_IDENTIFIER, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+				'meta_value' => $identifier, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+			]
+		);
+
+		if ( 0 === $query->post_count ) {
+			return null;
+		}
+
+		return $query->posts[0];
+	}
+
+	/**
+	 * Get all emails for type.
+	 *
+	 * @internal
+	 *
+	 * @param string $type Email type.
+	 * @param int    $limit Limit. Default 10.
+	 * @param int    $offset Offset. Default 0.
+	 * @return Email[]
+	 */
+	public function get_all( string $type, $limit = 10, $offset = 0 ): array {
+		return get_posts(
+			[
+				'post_type'      => Email_Post_Type::POST_TYPE,
+				'meta_key'       => self::META_TYPE, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+				'meta_value'     => $type, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+				'posts_per_page' => $limit,
+				'offset'         => $offset,
+			]
+		);
+	}
+
+	/**
+	 * Count all emails.
+	 *
+	 * @internal
+	 *
+	 * @return int
+	 */
+	public function count() {
+		$query = new WP_Query(
+			[
+				'post_type' => Email_Post_Type::POST_TYPE,
```

Comment:
A tiny optimization here could be the use of `posts_per_page` equals 1.

Reply:
Thanks, yeah. And rename the method to something like `is_empty` or `has_emails` and return `bool`.

Reply:
It should still return the correct count even with `posts_per_page` of 1 so you could leave the method name.